src/document.rs

```diff
@@ -526,8 +526,8 @@ impl Document {
         Ok(())
     }
 
-    /// Attempts to decode a `Document` from a byte stream.
-    pub fn decode<R: Read + ?Sized>(reader: &mut R) -> DecoderResult<Document> {
+    /// Attempts to deserialize a `Document` from a byte stream.
+    pub fn deserialize<R: Read + ?Sized>(reader: &mut R) -> crate::de::Result<Document> {
```

Comment:
If you go with my `serialize_to` suggestion above, I think this should be renamed to `deserialize_from` for parity.